rfc/mediachain-rfc-4.md

```diff
@@ -0,0 +1,328 @@
+update object on top of the current chain, while lookup retrieved the
+head of the chain for a canonical.
+
+In practice, inserts are idempotent and the chained update model only
+serves to constrain the system by imposing order of operations.
+By dropping ordering and utilizing domain specific identifiers, we can
+eliminate the distinction between insert and update.
+
+The write interface is thus an upsert method, which combines insert
+and update, and adds a statement to the set of objects associated with
+an identifier. If the system has no prior knowledge of the identifier,
+then it creates a new entry in the identifier mapping with a singleton
+set containing the object.
+
+The lookup method can be retained as a basic read operation, with a
+change at its return value. Since there are no chained updates any
+more, the method returns the known set of updates relating to an
+identifier. 
+
+### Namespace Partitioning
+
+In order to reflect the natural organization of data according to institution, industry, topic, etc, we introduce the concept of namespaces, which immediately give us several advantages. In terms of scaling, the partitioning must be semantic so as to allow peers to commit resources only in topics they are interested in. In terms of administrative control, the partitioning must be hierarchical so that we can easily delegate management and moderation to organizational stake holders, see "Moderation and Publishing Model" below.
+
+To satisfy these requirements, we've chosen to adopt a hierarchical structure similar to unix paths. Leaves in the hierarchy map to specific datasets and provide the primary entry points for writes, and namespaces higher in the hierarchy aggregate namespaces below and provide a progressively expanded read view of the data structure. Much like symlinks with unix paths, alternate views of the hierarchy are possible, namespaces may have multiple parents, and so on. This creates a semantically rich and expressive framework for querying and composition, see "Namespace-based Queries and Aggregation" below.
+
+### Moderation and Publishing Model
+
+Namespaces allow us fine grained control on publishing and
+participation level. At one end of the spectrum, we want to
+encourage public namespaces which allow lightweight permissionless
+participation. At the other end, we want to support curated namespaces
+to combat spam and allow authoritative sources and commercial media
+providers to maintain control of their own data.
+
+In order to implement this model, each participating peer has an
+associated identity corresponding to a public key. The identities
+are intended to be long-lived and eventually have an attached reputation
+score.
+
+All statements published in the system are signed with the key of the
+source of the statement. In order to publish in a moderated namespace,
+the originating peer must present a certificate which allows it to
```

Comment:
I think we need to at lest outline, as permissions is an essential part of the protocol. I don't intend to go into painstaking detail, just enough to support the narrative of statement publication.